tripal_chado/includes/tripal_chado.fields.inc

```diff
@@ -2828,6 +2735,128 @@ function tripal_chado_bundle_instances_info_linker(&$info, $entity_type, $bundle
 
 }
 
+/**
+ * Used to find all of the properties for a given table.
+ *  
+ * @param $table_name
+ *   The name of the base table.
+ * @param unknown $prop_table
+ *   The name of the property table.
+ * @param $type_table
+ *   The name of the table that contains the type specifier.
+ * @param $type_column
+ *   The name of the column that contains the type specifier.
+ *   
+ * @return
+ *   An array of cvterm objects for the properties to be added as fields.
+ */
+function tripal_chado_bundle_get_properties($table_name, $prop_table, $type_table, $type_column) {
+  
+  $tschema = chado_get_schema($table_name);
+  $schema = chado_get_schema($prop_table);
+  $tpkey = $tschema['primary key'][0];
+  $pkey = $schema['primary key'][0];
+  
+  $props = NULL;
+  
+  // Property tables can be a bit tricky because not all property types
+  // in the prop table are appropriate for each type of data.  Also som
+  // bundle types are resolved via a property.  So, we have to distinguish
+  // between these two cases.
+  $sql = '';
+  $args = array();
+  
+  // First, is this the case where all of the records in the table are
+  // of this type?  If so, then all properties apply
+  if (!$type_column) {
+    $sql = 'SELECT DISTINCT type_id FROM {' . $prop_table . '}';
+    $props = chado_query($sql, $args);
+  }
+  // Second, if this is the case where a content type is uniquely identified
+  // by a type_id value in the base table, then only properties associated
+  // with that type ID should be used.
+  else if ($type_column and !$type_table) {
+    $sql = "
+        SELECT DISTINCT P.type_id
+        FROM {" . $prop_table . "} P
+          INNER JOIN {" . $table_name . "} T on T.$tpkey = P.$tpkey
+        WHERE T.$type_column = :cvterm_id
+      ";
+    $args[':cvterm_id'] = $cvterm_id;
+    $props = chado_query($sql, $args);
+  }
+  // Third, if this is the case where a content type is uniquely identified
+  // via a term/value pair in the prop table.
+  else if ($type_column and $type_table == $prop_table and !empty($type_value)) {
+    $sql = "
+        SELECT DISTINCT P2.type_id
+        FROM {" . $prop_table . "} P1
```

Comment:
should this use `db_escape_table`?   https://api.drupal.org/api/drupal/includes%21database%21database.inc/function/db_escape_table/7.x

Reply:
yeah, agreed. I'll fix it

Reply:
Okay, I've added the updates to the code and tested it again to make sure I didn't flub anything up. Still works for me.  If you approve then please feel free to merge!

Reply:
ok thanks!